File: 2GII/PDOO/QytetetRuby/lib/titulo_propiedad.rb
#encoding: utf-8

require_relative "jugador"
require_relative "casilla"

module ModeloQytetet
  
  class TituloPropiedad
    attr_reader :nombre, :alquiler_base, :factor_revalorizacion, :hipoteca_base\
                , :precio_edificar
    attr_writer :propietario 
    attr_accessor :hipotecada, :casilla

    # -------------------------- Constructor -----------------------------------    
    def initialize (nombre, alquiler_base, factor_revalorizacion, hipoteca_base\
                          , precio_edificar)
      @nombre = nombre
      @alquiler_base = alquiler_base
      @factor_revalorizacion = factor_revalorizacion
      @hipoteca_base = hipoteca_base
      @precio_edificar = precio_edificar
      @hipotecada = false
      @propietario = nil
    end

    # ---------------------------- Consultores ---------------------------------
    def tengo_propietario
      (@propietario == nil) ? false : true
    end    

    def propietario_encarcelado
      @propietario.encarcelado
    end
        
    # ---------------------------- Métodos -------------------------------------
    def cobrar_alquiler(coste)
      @propietario.modificar_saldo(-coste)
    end
        
    def to_s()
      (@hipotecada == true)? hipotecada = "\n- Está hipotecada"\
                             : hipotecada = "\n- No está hipotecada"
        
      "- Nombre de la calle: " + @nombre + hipotecada + "\n- Hipoteca base: " + 
      @hipoteca_base.to_s + "\n- Alquiler base: "     + @alquiler_base.to_s + 
      "\n- Precio por edificar: "      + @precio_edificar.to_s +
      "\n- Factor de revalorización: " + @factor_revalorizacion.to_s
    end

  end
  
end

File: 2GII/PDOO/QytetetJava/src/modeloqytetet/TituloPropiedad.java
package modeloqytetet;

public class TituloPropiedad {
    private String  nombre;
    private Boolean hipotecada;
    private float   factorRevalorizacion;
    private int     alquilerBase,
                    hipotecaBase,
                    precioEdificar;
    
    private Jugador propietario = null;
    private Casilla casilla;
    
    // ------------------------- Constructor -----------------------------------
    public TituloPropiedad (String nombre, int alquilerBase, float factorRevalorizacion
                            , int hipotecaBase, int precioEdificar) {
        this.nombre = nombre;
        this.alquilerBase = alquilerBase;
        this.factorRevalorizacion = factorRevalorizacion;
        this.hipotecaBase = hipotecaBase;
        this.precioEdificar = precioEdificar;
        this.hipotecada = false;
    }
    
    // ------------------------- Consultores -----------------------------------
    public int getAlquilerBase() {
        return alquilerBase;
    }
    
    public float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }
    
    public int getHipotecaBase() {
        return hipotecaBase;
    }
    
    public Boolean getHipotecada() {
        return hipotecada;
    }
    
    public String getNombre() {
        return nombre;
    }

    public int getPrecioEdificar() {
        return precioEdificar;
    }
    
    public Casilla getCasilla() {
        return casilla;
    }
    
    Boolean tengoPropietario() {
        return propietario != null;
    }

    Boolean propietarioEncarcelado() {
        return propietario.getEncarcelado();
    }       
    
    // ------------------------- Modificadores ---------------------------------
    void setCasilla(Casilla casilla) {
        this.casilla = casilla;
    }
    
    void setHipotecada(Boolean hipotecada) {
        this.hipotecada = hipotecada;
    }
    
    void setPropietario(Jugador propietario) {
        this.propietario = propietario;   
    }
    
    // ---------------------------- Métodos ------------------------------------    
    void cobrarAlquiler(int coste) {
        propietario.modificarSaldo(-coste);
    }    
    
    @Override
    public String toString(){
        String hipotec = new String();
        if (hipotecada)
            hipotec = "\n· Está hipotecada";
        else
            hipotec = "\n· No está hipotecada";
        
        return "· Nombre de la calle: " + nombre + hipotec +
                "\n· Hipoteca base: "   + hipotecaBase + "\n· Alquiler base: " +
                alquilerBase + "\n· Precio por edificar: " + precioEdificar +
                "\n· Factor de revalorización: " + factorRevalorizacion;
    }
}
